tests/integration/test_etcd.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+import subprocess
+import tarfile
+from pathlib import Path, PosixPath
+from shutil import rmtree
+from urllib.request import urlretrieve
+
+import pytest
+from juju.application import Application
+from juju.unit import Unit
+from pytest_operator.plugin import OpsTest
+
+from .constants import (
+    APP,
+    DATA_INTEGRATOR,
+    ETCD,
+    TLS_CERTIFICATES_APP_NAME,
+)
+from .helpers import check_secrets_usage_matching_juju_version
+from .markers import only_on_localhost, only_with_juju_secrets
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.group(1)
+@only_on_localhost
+@only_with_juju_secrets
+@pytest.mark.abort_on_fail
+async def test_deploy(
+    ops_test: OpsTest, app_charm: PosixPath, data_integrator_charm: PosixPath, cloud_name: str
+):
+    """Deploys charms for testing."""
+    tls_config = {"ca-common-name": "CN_CA"}
+    model_config = {
+        "logging-config": "<root>=INFO;unit=DEBUG",
+    }
+    await ops_test.model.set_config(model_config)
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            ETCD[cloud_name],
+            channel="3.5/edge",
+            application_name=ETCD[cloud_name],
+            num_units=3,
+        ),
+        ops_test.model.deploy(TLS_CERTIFICATES_APP_NAME, channel="stable", config=tls_config),
```

Comment:
The TLS operator added tracks recently and libjuju can't find the correct channel, if the track is not specified. use `latest/stable` or `1/stable`, depending on which one is needed here, instead of just `stable`.